### PR TITLE
Document breaking changes for completion and context suggester

### PR DESCRIPTION
**Pull request by areek**
breaking changes for https://github.com/elastic/elasticsearch/pull/14410

### CODE REVIEW

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
+[[breaking_50_suggester]]
+=== Suggester changes
+
+Now `completion` and `context` suggest fields are indexed as lucene's
+suggest fields. This means new `completion` indices are not compatible
+with pre-5.0 indices. Older `completion` indices can still be queried
+and indexed in the same way as before but not created. It is strongly
+recommended to reindex `completion` fields to take advantage of the
+following:
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: I'd replace this para with:

    The completion suggester has undergone a complete rewrite. This means that the
    syntax and data structure for fields of type `completion` have changed, as
    have the syntax and response of completion suggester requests.

    For indices created before Elasticsearch 5.0.0, `completion` fields and the
    completion suggester will continue to work as they did in Elasticsearch 2.x.
    However, it is not possible to run a completion suggester query across indices
    created in 2.x and indices created in 5.x.

    It is strongly recommended to reindex indices containing 2.x `completion`
    fields in 5.x to take advantage of the new features listed below.

    NOTE: You will need to change the structure of the completion field values
    when reindexing.

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+==== Completion suggester is near-real time
+
+Previously, deleted suggestions could be included in results even
+after refreshing an index. Now, deletions are visible in near-real
+time, i.e. as soon as the index has been refreshed. This applies
+to suggestion entries for both context and completion suggesters.
+
+==== Completion suggester is document-oriented
+
+Suggestions are aware of the document they belong in. This enables
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: belong in -> belong to.

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+retrieving any field value from the document. This is exposed
+through query-time `payload` option in `completion` and `context`
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: through THE

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+       "completion": {
+         "field": "fooSuggest",
+         "payload": ["field1", "field2"]
+       }
+     }
+  }
+}
+---------------
+
+Previously, `context` and `completion` suggesters supported index-time
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: AN index-time

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+`payloads` option, which was used to return metadata with suggestions.
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: STORE AND return metadata

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+Now metadata can be stored as a field in the same document as the
+suggestion for enabling retrieval at query-time. The support for
+index-time `payloads` has been removed to avoid bloating the in-memory
+index with suggestion metadata. Note that query-time `payloads` retrieval
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: Perhaps delete the sentence "Note that..."  and replace it with: 

    The time that it takes to retrieve payloads depends heavily on the size of the `_source` field.  
    The smaller the `_source`, the faster the retrieval.

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+is slightly less performant than the previous index-time `payloads`
+option as they are extracted from the document source instead of being
+fetched from the in-memory `completion` index`.
+
+==== Simpler completion indexing
+
+As suggestions are document-oriented, suggestion metadata (e.g. `output`)
+can be stored in a separate field. The support for specifying `output`
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: can be stored in a separate field -> should now be specified as a field in the document

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+when indexing suggestion entries has been removed. Now suggestion result
+entry's `text` is always the un-analyzed value of the suggestion's `input`
+(same as not specifying `output` while indexing suggestions in pre-5.0
+indices). Note that suggestions can be still indexed with `output` option
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: Delete: "Note that..."  - we said at the top that old indices function as before.

File: docs/reference/migration/migrate_5_0/suggest.asciidoc [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+against `completion` indices that were created pre-5.0.
+
+==== Completion mapping with multiple contexts
+
+The `context` option in `completion` field mapping is now an array to support
+multiple named `context` per completion field. Note that this is sugar for
```
Comment (clintongormley) [inline docs/reference/migration/migrate_5_0/suggest.asciidoc]: `context` -> contexts

**Comment by clintongormley** [comment]
Small suggestions, other than that LGTM. Thanks for the good writeup

**Comment by areek** [comment]
Thanks @clintongormley for the review!

**Pull request merged**